service/simpleSecretService.go

```diff
@@ -1,37 +1,49 @@
 package service
 
 import (
-	"awesomeProject3/service/de_encoding"
+	"awesomeProject3/service/ciphering"
 	"awesomeProject3/storage"
 )
 
 type SimpleSecretService struct {
-	storage storage.Storage
-	encoder de_encoding.Encoder
+	storage          storage.Storage
+	encoder          ciphering.Encoder
+	notrandomencoder ciphering.Encoder
 }
 
-func (ss *SimpleSecretService) ReadSecret(key string) (string, error) {
-	encryptedVal, err := ss.storage.Read(key)
-	checkError(err)
-	return ss.encoder.Decrypt(encryptedVal), nil
+func (ss *SimpleSecretService) ReadSecret(key string, password string) (string, error) {
+	codedKey, err := ss.notrandomencoder.Encrypt(key, ciphering.PassToSecretKey(password)) //encode key without random aspect
```

Comment:
It will be better to a description to the constructor of secret service about why you need two encoders and what role they play. Pay attention that you should always write comments about WHY, not WHAT.
```suggestion
	codedKey, err := ss.notrandomencoder.Encrypt(key, ciphering.PassToSecretKey(password))
```